pkgs/tools/security/gnupg/21.nix

```diff
@@ -23,9 +23,14 @@ stdenv.mkDerivation rec {
   };
 
   buildInputs = [
-    pkgconfig libgcrypt libassuan libksba libiconv npth gettext texinfo
-    readline libusb gnutls adns openldap zlib bzip2
-  ];
+    pkgconfig libgcrypt libassuan libksba libiconv npth gettext texinfo ]
+      ++ optional (adns     != null) adns
```

Comment:
The null pattern is not so easy to understand - it wasn't immediately obvious to me as this is. Likely more time is spent reading than writing declarations, and this pattern is very clear. It is easy for people to understand that they can just 'flip' something at the top resulting in their own derivations and overlays. But my personal happiness doesn't depend on it - as you both like.